### PR TITLE
First pull request

### PR DESCRIPTION
**Pull request by LucasMRC**
- organized routes
- set up models
- set up mongodb
- created async error handler
- set up routes
- set up passport 
- set up authentication middleware

### CODE REVIEW

File: leaderboard/app.js [hunk 1]
```diff
@@ -1,39 +1,99 @@
...
 const path = require('path');
 const favicon = require('serve-favicon');
 const logger = require('morgan');
 const cookieParser = require('cookie-parser');
+const mongoose = require('mongoose');
+const engine = require('ejs-mate');
+const passport = require('passport');
+const session = require('express-session');
+const methodOverride = require('method-override');
+const User = require('./models/users');
```
Comment (nax3t) [inline leaderboard/app.js]: should be ./models/user

File: leaderboard/app.js [hunk 2]
```diff
@@ -1,39 +1,99 @@
...
   err.status = 404;
   next(err);
 });
 
-// error handler
+// error handler ===========================================
+
 app.use((err, req, res, next) => {
   // set locals, only providing error in development
+  req.user
```
Comment (nax3t) [inline leaderboard/app.js]: what's this line doing?

File: leaderboard/controllers/admin.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const User = require('../models/users');
```
Comment (nax3t) [inline leaderboard/controllers/admin.js]: change to user (singular)

File: leaderboard/controllers/index.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const Week = require('../models/weeks');
```
Comment (nax3t) [inline leaderboard/controllers/index.js]: change to week, singular

File: leaderboard/controllers/index.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+	},
+
+	/* GET about page. */
+	getAbout (req, res, next) {
+		res.render('about');
+	},
+
+	/* GET weeks page. */
+	async getWeeks (req, res, next) {
+		await Week.find({}, (err, weeks) => {
```
Comment (nax3t) [inline leaderboard/controllers/index.js]: no callback here, use `const weeks = await Week.find({});` and handle error with middleware or wrap in a try/catch block

File: leaderboard/controllers/weeks.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+	async postAdminWeek (req, res, next) {
+ 		let newWeek = {
+ 			dateRange: req.body.weekDays,
+ 			winners: [
+	 			{ place: '1st', name: req.body.first },
+	 			{ place: '2nd', name: req.body.second },
+	 			{ place: '3rd', name: req.body.third }
+ 			]
+ 		};
+ 		await Week.create(newWeek, (err, createdWeek) => {
```
Comment (nax3t) [inline leaderboard/controllers/weeks.js]: `const week = await Week.create(newWeek);`

File: leaderboard/controllers/weeks.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+ 			} else {
+ 				console.log(createdWeek);
+				res.redirect('/admin/weeks');
+ 			}
+ 		});	
+ 	},
+
+	/* GET /admin/weeks */
+	async getAdminWeeks (req, res, next) {
+		await Week.find({}, (err, weeks) => {
```
Comment (nax3t) [inline leaderboard/controllers/weeks.js]: `const weeks = await Week.find({});`

File: leaderboard/controllers/weeks.js [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+	 			{ place: '3rd', name: req.body.third }
+ 			];
+ 		week.dateRange = req.body.dateRange;
+	  week.save();
+	  res.redirect('/admin/weeks');
+	},
+
+	/* DELETE admin/weeks/:id */
+	async deleteWeek (req, res, next) {
+		await Week.findByIdAndRemove(req.params.id, (err) => {
```
Comment (nax3t) [inline leaderboard/controllers/weeks.js]: no callback here

File: leaderboard/middleware/index.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+	// Authentication middleware
+	isLoggedIn: (req, res, next) => {
+    if(req.isAuthenticated()){
+        return next();
+    }
+    res.redirect('/admin/login');
+	},
+	// Check if user is Admin
+	requireAdmin: (req, res, next) => {
```
Comment (nax3t) [inline leaderboard/middleware/index.js]: rename requireAdmin to isAdmin

File: leaderboard/models/users.js [hunk 10]
```diff
@@ -0,0 +1,11 @@
+const mongoose = require('mongoose');
```
Comment (nax3t) [inline leaderboard/models/users.js]: rename this file to user.js

File: leaderboard/models/weeks.js [hunk 11]
```diff
@@ -0,0 +1,13 @@
+const mongoose = require('mongoose');
```
Comment (nax3t) [inline leaderboard/models/weeks.js]: rename this file to week.js

**Review by nax3t — CHANGES_REQUESTED** [review]
Thanks for all the hard work!  Looking good. Please see my comments re: needed changes, re-submit changes as a new PR and I will merge them.

Thanks!

**Comment by LucasMRC** [comment]
- removed extra req.user 
- changed model names to singular
- changed requireAdmin to isAdmin
- set up error handling middleware

**Pull request closed without merging**